compiler/tests-compiler/effects_continuations.ml

```diff
@@ -120,21 +120,22 @@ let%expect_test "test-compiler/lib-effects/test1.ml" =
       }
       var m = 0, _y_ = 0;
      }
-     runtime.caml_push_trap
-      (function(_E_){
-        if(_E_ === Stdlib[8]) return cont(0);
-        var raise = caml_pop_trap();
-        return raise(caml_maybe_attach_backtrace(_E_, 0));
-       });
-     if(caml_string_equal(s, cst)){
+     if
+      (runtime.caml_push_trap
+        (function(_E_){
+          if(_E_ === Stdlib[8]) return cont(0);
+          var raise = caml_pop_trap();
+          return raise(caml_maybe_attach_backtrace(_E_, 0));
+         }),
+       caml_string_equal(s, cst)){
       var _z_ = Stdlib[8], raise = caml_pop_trap();
       return raise(caml_maybe_attach_backtrace(_z_, 1));
      }
      var _A_ = Stdlib[79];
      return caml_cps_call2
              (_A_,
               cst_toto,
-              function(_D_){caml_pop_trap(); return cont([0, [0, _D_, n, m]]);});
+              function(_D_){return caml_pop_trap(), cont([0, [0, _D_, n, m]]);});
```

Comment:
note that we don't save any space here.

Reply:
True, but Jérôme’s point in the issue was:

> Then, we may be able to coalesce more variable declarations, or omit the return of an arrow function.

So there can be a benefit. What’s not clear to me is whether these further compactions are already happening or whether they require further changes.

Reply:
With `--enable es6`, this really makes a difference, indeed.